app/components/channel_drawer/channel_drawer.js

```diff
@@ -279,6 +248,12 @@ export default class ChannelDrawer extends Component {
 
     onPageSelected = (index) => {
         this.swiperIndex = index;
+        if (this.drawerSwiper && this.swiperIndex !== 1 && this.props.teamsCount > 1) {
+            this.setState({locked: true});
```

Comment:
why are you locking the drawer when swiping to show the teams? if the drawer is closed when showing the teams it should reset the drawer to show the channels next time is opened

Reply:
without it the team/channel switcher won't be able to receive swipe events. have you checked have it works in practcie?

Reply:
it's always resetting properly, check it